client/src/components/pages/RobotInteractionCockpit/RobotInteractionCockpit.jsx

```diff
@@ -0,0 +1,167 @@
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable no-plusplus */
+import React, { useState, useEffect, useCallback, useRef } from 'react';
+import { Layout, Card, Steps, Space, Button, Typography } from 'antd';
+import HeaderNavbar from '../../content/HeaderNavbar/HeaderNavbar';
+import RobotInteractionInputSection from '../../content/RobotInteractionSections/RobotInteractionInputSection';
+import { getAllParametersForRobot } from '../../../api/variableRetrieval';
+import { isRobotExecutable } from '../../../utils/robotExecution';
+import { startRobotForUser } from '../../../api/socketHandler/socketEmitter';
+
+const { Step } = Steps;
+const { Title } = Typography;
+
+/**
+ * @description Page, where you can interact with a robot and for example enter input
+ * @category Client
+ * @component
+ */
+const RobotInteractionCockpit = (match) => {
+  const { robotId } = match.match.params;
+  const { userId } = match.location.state;
+  const isMounted = useRef(true);
+  const [parameterList, setParameterList] = useState([]);
+  const [currentStep, setCurrentStep] = useState(0);
+  const [parameters, setParameters] = useState([]);
+
+  /**
+   * @description For each activity of the current robot get the id, the name and all the parameters that require a user input
+   */
+  const getActivityAndParameterInformation = useCallback(() => {
```

Comment:
Why is the useCallback needed here?

Reply:
I think this [article](https://medium.com/@infinitypaul/reactjs-useeffect-usecallback-simplified-91e69fb0e7a3) explains it quite well